server/kubernetes.go

```diff
@@ -0,0 +1,144 @@
+package server
+
+import (
+	"context"
+	"encoding/base64"
+	"log"
+	"os"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/eks"
+	"github.com/mattermost/mattermost-cloud/k8s"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/pkg/errors"
+	logrus "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/aws-iam-authenticator/pkg/token"
+)
+
+// Deployment contains information needed to create a deployment in Kubernetes
+type Deployment struct {
+	Namespace      string
+	ImageTag       string
+	DeployFilePath string
+	Environment    CWS
+}
+
+func namespaceExists(kc *k8s.KubeClient, namespaceName string) (bool, error) {
+	_, err := kc.Clientset.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return false, err
+	} else if err != nil && k8sErrors.IsNotFound(err) {
+		return false, nil
+	}
+	return true, nil
+}
+
+func getOrCreateNamespace(kc *k8s.KubeClient, namespaceName string) (*corev1.Namespace, error) {
+	namespace, err := kc.Clientset.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+	if err != nil && k8sErrors.IsNotFound(err) {
+		return kc.CreateOrUpdateNamespace(namespaceName)
+	}
+
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return nil, err
+	}
+
+	return namespace, nil
+}
+
+func deleteNamespace(kc *k8s.KubeClient, nameSpaceName string) error {
+	namespace := []string{nameSpaceName}
+	err := kc.DeleteNamespaces(namespace)
+	if err != nil {
+		return err
```

Comment:
It's nice to return errors with more information with `errors.Wrap()`. Just a heads up for future PRs.